src/arkode/arkode_splittingstep_coefficients.c

```diff
@@ -0,0 +1,474 @@
+            SUNRpowerR(composition_stages - 1, SUN_RCONST(1.0) / (order - 1)));
+  for (int i = 1; i <= composition_stages; i++)
+  {
+    /* To avoid roundoff issues, this ensures end_cur=1 for the last value of i*/
+    const sunrealtype end_cur = 2 * i < composition_stages
+                                  ? (start + i * gamma)
+                                  : (end + (i - composition_stages) * gamma);
+    /* Recursively generate coefficients and shift beta_cur */
+    beta_cur  = SplittingStepCoefficients_ComposeStrangHelper(partitions,
+                                                              order - 2,
+                                                              composition_stages,
+                                                              start_cur, end_cur,
+                                                              beta_cur);
+    start_cur = end_cur;
+  }
+
+  return beta_cur;
+}
+
+/*---------------------------------------------------------------
+  Routine which does validation and setup before calling
+  SplittingStepCoefficients_ComposeStrangHelper to fill in the
+  beta coefficients
+  ---------------------------------------------------------------*/
+static SplittingStepCoefficients SplittingStepCoefficients_ComposeStrang(
+  const int partitions, const int order, const int composition_stages)
+{
+  if (order < 2 || order % 2 != 0)
+  {
+    // Only even orders allowed
+    return NULL;
+  }
+
+  const int stages = 1 + (partitions - 1) *
+                           SUNIpowerI(composition_stages, order / 2 - 1);
+  const SplittingStepCoefficients coefficients =
+    SplittingStepCoefficients_Alloc(1, stages, partitions);
+  if (coefficients == NULL) { return NULL; }
+
+  coefficients->order    = order;
+  coefficients->alpha[0] = SUN_RCONST(1.0);
+
+  SplittingStepCoefficients_ComposeStrangHelper(partitions, order,
+                                                composition_stages,
+                                                SUN_RCONST(0.0), SUN_RCONST(1.0),
+                                                coefficients->beta[0]);
+
+  return coefficients;
+}
+
+SplittingStepCoefficients SplittingStepCoefficients_TripleJump(const int partitions,
+                                                               const int order)
+{
+  return SplittingStepCoefficients_ComposeStrang(partitions, order, 3);
+}
+
+SplittingStepCoefficients SplittingStepCoefficients_SuzukiFractal(
+  const int partitions, const int order)
+{
+  return SplittingStepCoefficients_ComposeStrang(partitions, order, 5);
+}
+
+/*---------------------------------------------------------------
+  Routine to print a splitting coefficient structure
+  ---------------------------------------------------------------*/
+void SplittingStepCoefficients_Write(const SplittingStepCoefficients coefficients,
+                                     FILE* const outfile)
+{
+  // TODO(SBR): update when https://github.com/LLNL/sundials/pull/517 merged
```

Comment:
Comment to remind you of this `TODO(SBR)`

Reply:
#517 will be merged after this PR